### PR TITLE
C utils, use colnamesInt in R functions

### PR DESCRIPTION
**Pull request by jangorecki**
- move some (tiny or very new) C functions into `utils.c`
- in many R functions use C helper function `colnamesInt` that process a list of columns to integer, corresponding to a provided data.table. As a result we get:
1. reduce number of lines of code
2. unify handling of input (like checking `isRealReallyInt` on real input)
3. speed up code
4. as of now (for efficiency and lesser complexity) only first non-existing column is reported

----

I am pretty sure there are still some other places in R code that could use `colnamesInt` helper.
To do:  use it from C functions too.

### CODE REVIEW

File: R/data.table.R [hunk 1]
```diff
@@ -2505,28 +2495,16 @@ setnames = function(x,old,new,skip_absent=FALSE) {
 
 setcolorder = function(x, neworder=key(x))
 {
-  if (anyDuplicated(neworder)) stop("neworder contains duplicates")
```
Comment (MichaelChirico) [inline R/data.table.R]: this was a different error right? `setcolorder(x, c(1, 1, 2, 3))`?

Comment (jangorecki) [reply R/data.table.R]: Yes, many error messages got changed as they are now handled in single place

Comment (MichaelChirico) [reply R/data.table.R]: I meant the test was for different behavior?

Comment (jangorecki) [reply R/data.table.R]: these tests duplicates on `names(x)`, not `neworder` argument which is checked inside `colnamesInt`. Check for duplicates in `names(x)` was made later in `if (is.character(neworder))` branch.

**Comment by mattdowle** [comment]
Very nice! It was surprising to see how much duplication of logic there was here. Much better being in one place now.

**Pull request merged**